File: Task 03/python/Subtask4/Task4.py
with open('input.txt', 'r') as infile:
    n = int(infile.read().strip())

lines = []
for i in range(n):
    lines.append(' ' * (n - i - 1) + '*' * (2 * i + 1))

for i in range(n - 2, -1, -1):
    lines.append(' ' * (n - i - 1) + '*' * (2 * i + 1))

with open('output.txt', 'w') as outfile:
    outfile.write('\n'.join(lines))

File: Task 03/elixir/Subtask4/Task4.ex
dat=File.read("Input.txt") |> String.trim()
n=String.to_integer(dat)

u=
  for i<-0..(div(n,2))do
    space=div(n-2*i-1,2)
    ast=2*i+1
    String.duplicate(" ",space)<>String.duplicate("*",ast)<> "\n"
  end
  
l=
  for i<-(div(n,2)-1)..0 do
    space=div(n-2*i-1,2)
    ast=2*i+1
    String.duplicate("",space)<>String.duplicate("*",ast)<> "\n"
  end
dia=Enum.join(u+l)
File.write("output.txt",dia)
end
